llvm/lib/Support/SpecialCaseList.cpp

```diff
@@ -143,63 +177,55 @@ SpecialCaseList::addSection(StringRef SectionStr, unsigned LineNo,
   return &Section;
 }
 
-bool SpecialCaseList::parse(const MemoryBuffer *MB, std::string &Error) {
-  Section *CurrentSection;
-  if (auto Err = addSection("*", 1).moveInto(CurrentSection)) {
-    Error = toString(std::move(Err));
-    return false;
-  }
-
+llvm::Expected<ParsedSpecialCaseList>
+ParsedSpecialCaseList::parse(const MemoryBuffer &MB) {
+  ParsedSpecialCaseList Result;
   // In https://reviews.llvm.org/D154014 we added glob support and planned to
   // remove regex support in patterns. We temporarily support the original
   // behavior using regexes if "#!special-case-list-v1" is the first line of the
   // file. For more details, see
   // https://discourse.llvm.org/t/use-glob-instead-of-regex-for-specialcaselists/71666
-  bool UseGlobs = !MB->getBuffer().starts_with("#!special-case-list-v1\n");
-
-  for (line_iterator LineIt(*MB, /*SkipBlanks=*/true, /*CommentMarker=*/'#');
+  Result.UseRegexes = MB.getBuffer().starts_with("#!special-case-list-v1\n");
```

Comment:
Should it be called UseTegexes or UseGlobs?
Can we make this change orthogonal to this PR?